consensus/raft/raft.go

```diff
@@ -686,3 +695,47 @@ func find(s []string, elem string) bool {
 	}
 	return false
 }
+
+func (rw *raftWrapper) observePeers() {
+	obsCh := make(chan hraft.Observation, 1)
+	defer close(obsCh)
+
+	if !sixtyfour {
+		// 32-bit systems don't support observers
+		// TODO: Check if this is still the case?
+		logger.Warning("raft Observers panic on 32-bit architecture")
+		return
+	}
+
+	observer := hraft.NewObserver(obsCh, false, nil)
+	rw.raft.RegisterObserver(observer)
+	defer rw.raft.DeregisterObserver(observer)
+
+	for {
+		select {
+		case obs := <-obsCh:
+			switch obs.Data.(type) {
+			case hraft.PeerObservation:
+				pObs := obs.Data.(hraft.PeerObservation)
+				if pObs.Removed {
+					logger.Infof("Observed removal of raft peer: %s",
```

Comment:
we have a tendency to start log messages with lowercase. (golint complains for Printlns). Message could be a bit more informative: `raft peer departed. Removing from peerstore`.